File: Stylpic/Stylpic/SPUser.m
//
//  SPUser.m
//  Stylpic
//
//  Created by Joshua Bell on 3/22/15.
//  Copyright (c) 2015 Neil Bhargava. All rights reserved.
//

#import "SPUser.h"
#import <Parse/PFObject+Subclass.h>

@implementation SPUser

@synthesize isFollowing;

@dynamic firstName, lastName, profilePicture, facebookId, facebookFriends, channel;

+ (void)load {
    [self registerSubclass];
}

- (NSString *)spDisplayName {
    //NSString *lastInitial = [self.lastName substringToIndex:1];
    return [[NSString stringWithFormat:@"%@%@", self.firstName, self.lastName] lowercaseString];
}


@end

File: Stylpic/Stylpic/SPUser.swift
//
//  SPUser.swift
//  Stylpic
//
//  Created by Neil Bhargava on 3/1/15.
//  Copyright (c) 2015 Neil Bhargava. All rights reserved.
//

import UIKit

class SPUser: PFUser, PFSubclassing {
    @NSManaged var displayName : String
    @NSManaged var profilePicture : PFFile
    @NSManaged var facebookId : String
    @NSManaged var facebookFriends : [SPUser]
    @NSManaged var channel : String
    @NSManaged var isFollowing : Bool
    @NSManaged var firstName : String
    @NSManaged var lastName : String
    
    override class func initialize() {
        var onceToken : dispatch_once_t = 0;
        dispatch_once(&onceToken) {
            self.registerSubclass()
        }
    }
    
    func spDisplayName() -> String {
        var lastInitial = (self.lastName as NSString).substringToIndex( 1 )
        return "\(self.firstName) \(lastInitial)."
    }
}
